explainaboard/visualizers/draw_charts.py

```diff
@@ -0,0 +1,265 @@
+    elif len(sys_names) != len(reports):
+        raise ValueError('Length of sys_names must equal that of reports')
+
+    report_info: list[SysOutputInfo] = []
+    for report in reports:
+        with open(report) as fin:
+            report_info.append(SysOutputInfo.from_dict(json.load(fin)))
+
+    # --- Overall results
+    num_levels = len(unwrap(report_info[0].analysis_levels))
+    for level_id in range(num_levels):
+        overall_results: list[list[Performance]] = [
+            list(unwrap(x.results.overall)[level_id]) for x in report_info
+        ]
+        overall_metric_names = [x.metric_name for x in overall_results[0]]
+
+        ys = [[x.value for x in y] for y in overall_results]
+        y_errs = None
+        if overall_results[0][0].confidence_score_low is not None:
+            y_errs = [
+                (
+                    [x.value - unwrap(x.confidence_score_low) for x in y],
+                    [unwrap(x.confidence_score_high) - x.value for x in y],
+                )
+                for y in overall_results
+            ]
+
+        make_bar_chart(
+            ys,
+            output_dir,
+            'overall',
+            output_fig_format='png',
+            fig_size=(8, 6),
+            sys_names=sys_names,
+            errs=y_errs,
+            title=None,
+            xticklabels=overall_metric_names,
+            ylabel='metric value',
+        )
+
+    # --- analysis results
+    analysis_results: list[list[AnalysisResult]] = [
+        unwrap(x.results.analyses) for x in report_info
+    ]
+    if any([len(x) != len(analysis_results[0]) for x in analysis_results]):
+        raise ValueError(
+            f'mismatched number of analyses: {[len(x) for x in analysis_results]}'
+        )
+
+    # Bucket performance: feature name, for example, sentence length
+    for analysis_result in progress(zip(*analysis_results)):
+        if any([x.name != analysis_result[0].name for x in analysis_result]):
+            raise ValueError(
+                f'mismatched analyses: {[x.name for x in analysis_result]}'
+            )
+
+        if all([isinstance(x, BucketAnalysisResult) for x in analysis_result]):
+            plot_buckets(analysis_result, output_dir, sys_names)
+        elif all([isinstance(x, ComboCountAnalysisResult) for x in analysis_result]):
+            plot_combo_counts(analysis_result, output_dir, sys_names)
+        else:
+            raise ValueError('illegal types of analyses')
+
+
+def main():
+
+    parser = argparse.ArgumentParser(
+        description='Draw Histogram for ExplainaBoard Report'
+    )
+
+    parser.add_argument('--task', type=str, required=False, help="the task name")
+
+    parser.add_argument(
+        '--reports',
+        type=str,
+        required=True,
+        nargs="+",
+        help="the directories of reports. Multiple one should be separated by space, "
+        "for example: report1 report2",
+    )
+
+    parser.add_argument(
+        '--sys_names',
```

Comment:
How about following GNU guideline for long options?

> Long options consist of -- followed by a name made of alphanumeric characters and dashes. Option names are typically one to three words long, with **hyphens to separate words**.

https://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html

Reply:
This seems reasonable, but currently the whole project is standardized to underbars so we should probably change throughout the project if we want to do this. I opened an issue: https://github.com/neulab/ExplainaBoard/issues/390